components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts

```diff
@@ -628,27 +689,24 @@ export const useSwap = () => {
     }
 
     return getLocale('braveSwapReviewOrder')
-  }, [
-    selectedAccount,
-    fromToken,
-    swapValidationError,
-    selectedNetwork,
-    getLocale
-  ])
+  }, [fromToken, swapValidationError, selectedNetwork, getLocale])
 
   const isSubmitButtonDisabled = useMemo(() => {
     return (
+      !selectedNetwork ||
+      !selectedAccount ||
+      selectedNetwork.coin !== selectedAccount.accountId.coin ||
```

Comment:
That works for ETH and SOL, but doesn't in general. So for consistency if we need to match account with network let's use `networkSupportsAccount` from network-utils.ts which also works for other coins where accounts for different networks(testnet/mainnet) are incompatible.
@onyb

Reply:
Nice catch! I'll fix it in a subsequent PR.